### PR TITLE
Improve location parsing

### PR DESCRIPTION
**Pull request by rgrinberg**
Handle more formats, alerts, and format messages better.

### CODE REVIEW

File: otherlibs/ocamlc_loc/src/lexer.mll [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    match String.split_on_char '-' s with
+    | [x; y] -> (int_of_string x, int_of_string y)
+    | _ -> assert false
+}
+
+let blank = [' ' '\t' '\r']*
+
+let newline = '\n'
+
+let quoted_string = '"' [^ '"']+ '"'
```
Comment (Alizter) [inline otherlibs/ocamlc_loc/src/lexer.mll:46]: This negated pattern might end up swallowing new lines, if you care about that sort of thing.

File: otherlibs/ocamlc_loc/src/lexer.mll [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  | eof { Eof }
+
+and toplevel_message = parse
+  | blank ([^ '\n']* as message) blank '\n' { message }
+  | "" { "" }
+
+and token = parse
+  | (blank as indent) "File \"" ([^ '"']* as path) "\", " blank
+    (("line " (digits as line) | "lines " (range as lines)))
+    ("," blank "characters" blank (range as chars))?
```
Comment (Alizter) [inline otherlibs/ocamlc_loc/src/lexer.mll:92]: I don't know if OCaml does this, but if there is a 1 char range, does it default to `character`?

File: otherlibs/ocamlc_loc/src/lexer.mll (hunk 1 above)
Comment (rgrinberg) [reply otherlibs/ocamlc_loc/src/lexer.mll:46]: I suppose it's alright. The bigger problem is that we don't support quotes as part of the filename. But that's fine with me :)

File: otherlibs/ocamlc_loc/src/lexer.mll (hunk 2 above)
Comment (rgrinberg) [reply otherlibs/ocamlc_loc/src/lexer.mll:92]: I checked the ocaml source and `character` is indeed impossible.

**Review by Alizter — APPROVED** [review]

**Pull request merged**